src/transports/ws/sws.c

```diff
@@ -773,14 +744,21 @@ static int nn_sws_fail_conn (struct nn_sws *self, int code, char *reason)
 
     self->fail_msg_len += payload_len;
 
-    self->instate = NN_SWS_INSTATE_FAILING;
+    self->instate = NN_SWS_INSTATE_CLOSED;
 
-    /*  On the next recv, the connection will be failed. Why defer
-        until the next recv? Semantically, until then, this incoming
-        message has not been interpreted, so it's not until then that
-        it could be failed. This type of pre-processing is necessary
-        to early fail chunked transfers. */
```

Comment:
A more direct way of talking about this modification -- did you change it in order to simplify what seemed "wonky" or "wrong", or was it specifically to address a characterized failure? Don't take this question as accusatory -- rather, it's highlighting my own immediate lack of ability to give positive, helpful review of this change, without yet having mentally spooled up again!